### DISCUSSION TITLE
Nonlinear Convergence problem in Phase field

### DISCUSSION THREAD

**Discussion by ada-ayy** [idaholab/moose#17310] [Q&A Modules: Phase field] [answered]
Dear Moose users:

I am trying to solve an equation like this:
![equation](https://user-images.githubusercontent.com/73642707/111057691-6baeb400-84c4-11eb-92cd-6687cee311c6.png)
When I set the L to be a constant, it can be well solved. 
However, when I try to set L as a function of G (elastic energy, while Gc is the constant critical value), the time step keeps reducing by half. Even if I construct L to be a smooth function of G, it still faces the same problem.

I have tried to use ParsedMaterial to define the L.
Also, I have tried to use AuxVariable/ AuxKernel as below:
```
[AuxVariables]
 [./mob]
  order = CONSTANT
  family = MONOMIAL
 [../]
[]

[AuxKernels]
 [./MOB]
  type =MaterialRealAux
  property = Fe
  variable = mob
 [../]
[]

  [./L]
    type = ParsedMaterial
    f_name = L
    args = 'mob'
    function = 'if (mob <0.0001, 0, if( mob > 0.002,10,(mob-0.0001)*(mob-0.0001)*(mob-0.002)*(mob-0.002)))'
    outputs = exodus
    block = 0
  [../]
```
Thanks,
Ada

**Comment by GiudGiud** [comment, marked as answer]
Hi Ada

By the time step keeps reducing by half you mean the non-linear solves are not converging?
Can you plot the values of L in the domain by converting it to an AuxVariable using a MaterialAuxKernel to see what is happening with L
I reformatted your code with ``` btw. 

Guillaume

**Reply by ada-ayy** [reply]
Thank you for your reply!
This is the distribution of L at t = 0
![L-initial](https://user-images.githubusercontent.com/73642707/111579433-f9eea700-87f0-11eb-8cc4-a66a4555ed98.png)
and this is L at t = 3 (when it gets stuck)
![final-L](https://user-images.githubusercontent.com/73642707/111579485-0ecb3a80-87f1-11eb-9969-a746ae05718a.png)
![L-detail](https://user-images.githubusercontent.com/73642707/111579489-1094fe00-87f1-11eb-8be9-916ab35ab980.png)

Also, this is the elastic energy Fe at t=3
![Fe-final](https://user-images.githubusercontent.com/73642707/111579609-433ef680-87f1-11eb-9897-f67223511f8e.png)
By contrast, this is Fe when setting L = constant at 3
![Fe-L0](https://user-images.githubusercontent.com/73642707/111579636-5356d600-87f1-11eb-95bd-450c23dc517f.png)

**Reply by laagesen** [reply]
A zero mobility such as this is likely to be a problem when using an implicit time stepping scheme. You could try making your L some very small number like 10^-2 or 10^-3 such that microstructural evolution in the region where G<G_c is negligible. (This is probably more realistic physically anyway because it's unlikely that the mobility of a solute species is identically 0.)  You could also try using an explicit time stepper.

**Reply by ada-ayy** [reply]
Thank you for your advice!! I have used **'tanh'** as well to solve this problem.
